### ISSUE TITLE
How to get/set the Caret Position?

### ISSUE THREAD

**Issue by Naicos** [jacobslusser/ScintillaNET#386]
I'm working at a XML Editor with Search and GoToLine. Most parts are already implemented but i cannot find a way to set the Caret position for both.
Also is there a way to include a XSD File for autocompletition/intellisense?

**Comment by Aghlara** [comment]
Use property `CurrentPosition` for getting the current position. And call method `GotoPosition` to set a position.

**Comment by Naicos** [comment]
With GoToPosition(LineNumber) the Caret gets set only to the character of the given number (For example if my text has 106 Lines and I want to do to Line 50 im in Line 2 and character 50)

**Comment by Aghlara** [comment]
Sorry I don't understand you. Method GotoPosition accepts the character position not a line number.

**Comment by Naicos** [comment]
Yes but I want to set the Caret to a specific Line with my own GoToLine Dialog

**Comment by Aghlara** [comment]
You can look at `Lines` property which each of them has the `Position` and then use `GoToPosition`.

**Comment by Stumpii** [comment]
I thought:

MyScintilla.Lines[23].Goto();

or

MyScintilla.GotoPosition(MyScintilla.Lines[23].Position);

**Comment by jacobslusser** [comment]
@Stumpii and @Aghlara are correct. If you want to get to a specific character on a specific line, do the math that would determine what position that is, e.g. `line position start + offset`. 

However, I might suggest that you're making it unnecessarily hard for yourself. Why isn't your search/find dialog giving you the document position directly instead of the line+offset? If it did that you could pass the value straight to `GoToPosition`.
